test/tools/test_component_tool.py

```diff
@@ -0,0 +1,549 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
```

Comment:
Let's add tests for:
- `to_dict`/`from_dict`
- complete YAML serde ([here](https://github.com/deepset-ai/haystack/blob/3f15f38c51e420f49758e94bdf0e27447aae8520/test/components/tools/test_tool_invoker.py#L223) is a good example)